File: BIONOVA/bionova-backend/scratch/db_check.js
const { Client } = require('pg');

const client = new Client({
  connectionString: "postgresql://postgres.daaoeapbouspxcuprsqx:Atirath%402026@aws-1-ap-southeast-1.pooler.supabase.com:6543/postgres?pgbouncer=true"
});

async function run() {
  try {
    await client.connect();
    const res = await client.query(`
      SELECT column_name, data_type 
      FROM information_schema.columns 
      WHERE table_name = 'task_live_master';
    `);
    console.log("Columns in task_live_master:");
    for (let row of res.rows) {
      console.log(`  ${row.column_name} (${row.data_type})`);
    }
  } catch (err) {
    console.error(err);
  } finally {
    await client.end();
  }
}

run();

File: BIONOVA/bionova-backend/scratch/DbCheck.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.ResultSetMetaData;
import java.sql.Statement;

public class DbCheck {
    public static void main(String[] args) {
        String url = "jdbc:postgresql://aws-1-ap-southeast-1.pooler.supabase.com:6543/postgres?pgbouncer=true&prepareThreshold=0";
        String user = "postgres.daaoeapbouspxcuprsqx";
        String password = "Atirath@2026";
        try (Connection conn = DriverManager.getConnection(url, user, password);
             Statement stmt = conn.createStatement();
             ResultSet rs = stmt.executeQuery("SELECT * FROM task_live_master LIMIT 1")) {
            ResultSetMetaData meta = rs.getMetaData();
            int cols = meta.getColumnCount();
            System.out.println("Columns in task_live_master:");
            for (int i = 1; i <= cols; i++) {
                System.out.println("  " + meta.getColumnName(i) + " (" + meta.getColumnTypeName(i) + ")");
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
